pkg/testing/multipass/provisioner.go

```diff
@@ -170,6 +182,56 @@ func (p *provisioner) launch(ctx context.Context, cfg runner.Config, batch runne
 	return nil
 }
 
+func (p *provisioner) ensureInstanceNotExist(ctx context.Context, batch runner.OSBatch) error {
+	var output bytes.Buffer
+	proc, err := process.Start("multipass",
+		process.WithContext(ctx),
+		process.WithArgs([]string{"list", "--format", "json"}),
+		process.WithCmdOptions(
+			runner.AttachOut(&output),
+			runner.AttachErr(&output)))
+	if err != nil {
+		return fmt.Errorf("multipass list failed: %w", err)
+	}
+
+	<-proc.Wait()
```

Comment:
Shouldn't the exit code of the command be tested before we try to parse the output ? Shouldn't we do something if the command fails ?
(Same thing on line 226)